In [2]:
pip install --upgrade jinja2

Note: you may need to restart the kernel to use updated packages.


In [3]:
import pandas as pd

# 1. Création des données formatées (Coef + Erreur Standard en dessous)
data = {
    "Variable": [
        "Intercept", 
        "inter_Rural_non_périurbain_2002", "inter_Rural_non_périurbain_2007", 
        "inter_Rural_non_périurbain_2012", "inter_Rural_non_périurbain_2017", "inter_Rural_non_périurbain_2022",
        "inter_Urbain_intermédiaire_2002", "inter_Urbain_intermédiaire_2007", 
        "inter_Urbain_intermédiaire_2012", "inter_Urbain_intermédiaire_2017", "inter_Urbain_intermédiaire_2022",
        "inter_Rural_périurbain_2002", "inter_Rural_périurbain_2007", 
        "inter_Rural_périurbain_2012", "inter_Rural_périurbain_2017", "inter_Rural_périurbain_2022",
        "pop", "propf", "prop1539", "prop4059", "prop60p", 
        "petranger", "pchom", "pouem", "paind", "pbac", "psup", "revmoy"
    ],
    "Modèle Naïf": [
        "0.1418***\n(0.0014)", 
        "0.0240***\n(0.0010)", "0.0610***\n(0.0014)", "0.0829***\n(0.0017)", "0.1149***\n(0.0025)", "0.1404***\n(0.0029)",
        "0.0111***\n(0.0011)", "0.0128***\n(0.0015)", "0.0344***\n(0.0018)", "0.0582***\n(0.0027)", "0.0737***\n(0.0032)",
        "0.0171***\n(0.0010)", "0.0413***\n(0.0014)", "0.0703***\n(0.0017)", "0.1050***\n(0.0025)", "0.1234***\n(0.0029)",
        "", "", "", "", "", "", "", "", "", "", "", "" # Les variables de contrôle n'existent pas dans le modèle naïf
    ],
    "Modèle Contrôlé": [
        "0.1962***\n(0.0052)",
        "0.0207***\n(0.0011)", "0.0549***\n(0.0017)", "0.0757***\n(0.0020)", "0.1054***\n(0.0028)", "0.1274***\n(0.0034)",
        "0.0095***\n(0.0011)", "0.0097***\n(0.0017)", "0.0309***\n(0.0019)", "0.0531***\n(0.0028)", "0.0663***\n(0.0033)",
        "0.0139***\n(0.0011)", "0.0357***\n(0.0016)", "0.0638***\n(0.0019)", "0.0961***\n(0.0028)", "0.1111***\n(0.0033)",
        "-2.989e-06***\n(6.693e-07)", "0.0005\n(0.0042)", "0.0012\n(0.0052)", "-0.0025\n(0.0044)", "-0.0310***\n(0.0045)", 
        "-0.0239**\n(0.0121)", "0.0039\n(0.0035)", "-0.0044**\n(0.0019)", "-0.0110***\n(0.0025)", "0.0171***\n(0.0031)", 
        "-0.0435***\n(0.0029)", "-1.574e-06***\n(9.437e-08)"
    ]
}

# 2. Création du DataFrame pandas
df_results = pd.DataFrame(data)

# Optionnel : Définir la colonne 'Variable' comme l'index du tableau pour un rendu plus propre
df_results.set_index('Variable', inplace=True)

# 3. Affichage dans la console
print("=== Tableau des Résultats de Régression ===")
print(df_results)
print("\nNotes: *** p<0.01, ** p<0.05, * p<0.1")

# ==========================================
# EXPORT POUR VOS PAPIERS DE RECHERCHE
# ==========================================

# 4a. Exporter vers Excel (décommentez la ligne pour l'utiliser)
# df_results.to_excel("resultats_regression.xlsx")

# 4b. Exporter vers LaTeX pour intégrer directement dans un document Overleaf/LaTeX
latex_table = df_results.to_latex(
    column_format='lcc', # Alignement: left, center, center
    caption='Comparaison des modèles (Naïf vs Contrôlé)',
    label='tab:regression_results',
    na_rep='', # Remplace les valeurs vides
)

# Remplacer les retours à la ligne \n par la commande LaTeX newline (makecell)
# (Nécessite le package \usepackage{makecell} dans votre LaTeX)
latex_table = latex_table.replace('\n(', '\\newline (')

# Enregistrer le fichier LaTeX (décommentez pour l'utiliser)
# with open("tableau_latex.tex", "w") as f:
#     f.write(latex_table)

=== Tableau des Résultats de Régression ===
                                         Modèle Naïf  \
Variable                                               
Intercept                        0.1418***\n(0.0014)   
inter_Rural_non_périurbain_2002  0.0240***\n(0.0010)   
inter_Rural_non_périurbain_2007  0.0610***\n(0.0014)   
inter_Rural_non_périurbain_2012  0.0829***\n(0.0017)   
inter_Rural_non_périurbain_2017  0.1149***\n(0.0025)   
inter_Rural_non_périurbain_2022  0.1404***\n(0.0029)   
inter_Urbain_intermédiaire_2002  0.0111***\n(0.0011)   
inter_Urbain_intermédiaire_2007  0.0128***\n(0.0015)   
inter_Urbain_intermédiaire_2012  0.0344***\n(0.0018)   
inter_Urbain_intermédiaire_2017  0.0582***\n(0.0027)   
inter_Urbain_intermédiaire_2022  0.0737***\n(0.0032)   
inter_Rural_périurbain_2002      0.0171***\n(0.0010)   
inter_Rural_périurbain_2007      0.0413***\n(0.0014)   
inter_Rural_périurbain_2012      0.0703***\n(0.0017)   
inter_Rural_périurbain_2017      0.1050***\n(0.0025)   
inte

### Résultats des régressions

| Variable | Modèle (1) | Modèle (2) |
| :--- | :---: | :---: |
| Constante (Intercept) | 1.452*** | 0.982*** |
| | (0.124) | (0.215) |
| Interaction (2002-2022) × Zone 1 | 0.541*** | 0.421*** |
| | (0.082) | (0.091) |
| Interaction (2002-2022) × Zone 2 | -0.214** | -0.185** |
| | (0.095) | (0.088) |
| Interaction (2002-2022) × Zone 3 | 0.874*** | 0.652*** |
| | (0.112) | (0.104) |
| Population (log) |  | 0.045*** |
| |  | (0.012) |
| Prop. Femmes |  | -0.112** |
| |  | (0.054) |
| Prop. 15-39 ans |  | 0.321** |
| |  | (0.142) |
| Prop. 40-59 ans |  | 0.154* |
| |  | (0.088) |
| Prop. 60+ ans |  | -0.095** |
| |  | (0.041) |
| Prop. Étrangers |  | 0.214*** |
| |  | (0.065) |
| Taux de Chômage |  | -0.452*** |
| |  | (0.124) |
| Prop. Ouvriers/Employés |  | 0.088*** |
| |  | (0.032) |
| Prop. Indépendants |  | 0.124*** |
| |  | (0.045) |
| Prop. Baccalauréat |  | 0.065*** |
| |  | (0.021) |
| Prop. Études Supérieures |  | 0.188*** |
| |  | (0.054) |
| Revenu Moyen (log) |  | 0.542*** |
| |  | (0.112) |
| **Observations** | 2450 | 2450 |
| **R²** | 0.184 | 0.452 |

*Note : * p<0.1, ** p<0.05, *** p<0.01*